src/subcommands/subcommand.go

```diff
@@ -412,3 +417,23 @@ func executeStartServer(args []string) {
 		log.Fatalf("failed to serve: %v", err)
 	}
 }
+
+func convertToAbsPath(relPath string) string {
```

Comment:
you should only call this when sourceType is file

Reply:
it checks if the sourcetype is file in the function, are you suggesting to do that before the function? i thought it would be easier if everything was encapsulated in this function

Reply:
Yes, because it's not clear why we'd convert eg a website's url to an absolute path. It's more clear that we'd validate that the string is expected to be a path first, and then convert it to an absolute path. Also, we'll probably ahve to do similar kinds of normalization for all source types eventually, so the overall code structure will be more understandable if we switch by source type and then run the normalization for each separately.

Reply:
ok fixed